tracer/src/Datadog.Trace/AppSec/InstrumentationGateway.cs

```diff
@@ -30,13 +30,20 @@ internal partial class InstrumentationGateway
 
         public event EventHandler<InstrumentationGatewaySecurityEventArgs> RequestEnd;
 
+        public event EventHandler<InstrumentationGatewaySecurityEventArgs> BodyAvailable;
+
         public event EventHandler<InstrumentationGatewayEventArgs> LastChanceToWriteTags;
 
-        public void RaiseRequestStart(HttpContext context, HttpRequest request, Span relatedSpan, RouteData routeData) => RaiseEvent(context, request, relatedSpan, routeData, RequestStart);
+        public void RaiseRequestStart(HttpContext context, HttpRequest request, Span relatedSpan, RouteData routeData) =>
+            RaiseEvent(context, request, relatedSpan, routeData, null, RequestStart);
```

Comment:
I'd rather use named parameters everywhere you set a param to null

Reply:
> It seems very expensive to be reading the body using reflection every time, to then just be loading converting it back into a dictionary. Can we not use buffering to read the body directly as a json/form and use that directly. The buffering is a bit of a pain, but still seems like it would be preferable to this?
@andrewlock So we decided to go down that route for a first version, but were planning to improve / change the body extracting part later. The numbers aren't too bad for the moment: 
![image](https://user-images.githubusercontent.com/8353486/159382396-ef4591a3-2ece-484e-b0f0-4c7d8aa53c40.png)

Reply:
thanks fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2495/commits/e4116d0b92f32d2610da03b6dca0ea4ac1759a6a